src/main/scala/com/softwaremill/realworld/users/UsersRepository.scala

```diff
@@ -37,7 +37,7 @@ class UsersRepository(quill: Quill.Sqlite[SnakeCase]):
     } yield ur
   )
     .map(_.headOption)
-    .map(_.map(toUserDataWithPassword))
+    .map(_.map(UserWithPassword.fromRow))
```

Comment:
**issue:** If I understand correctly, `UserRow` is specific to the persistence layer. It shouldn't be a part of the `model`, but rather something separated in a `infrastructure` or `persistence` package with its specific Quill repository and conversion from `UserRow` to a domain (model) `UserData` type. This brings more question regarding the package layout used in this app. Do you want to preserve it to adhere to some standard convention for the "real-world" project, or maybe it could be refactored into a better structore?

Reply:
Good point! :) We can document the package structure as an ADR (separate task), and refactor if needed (also separate task). The real-world project assumptions only concern the API/domain, the implementation should follow whatever best practices we choose

Reply:
I like the idea of ADR :+1: @wydra98 @bartma11 you can create a separate PR with an `adr` folder, and the first "decision file", called `00-packages.md`, describing why and how you would like to restructure the packages and layers. The file should list context, rationale, decision and consequences. Example: https://github.com/joelparkerhenderson/architecture-decision-record/blob/main/examples/react-front-end-javascript-library/index.md

Reply:
I agree this is good point, I had same concern regarding returning UserRow from repository to the service - that's why there is following [TODO](https://github.com/softwaremill/realworld-tapir-zio/blob/master/src/main/scala/com/softwaremill/realworld/users/UsersRepository.scala#L24).
It should be removed along with other changes.